pkg/controllers/managementuser/rkecontrolplanecondition/rkecontrolplancondition.go

```diff
@@ -0,0 +1,143 @@
+	provisioningcontrollers "github.com/rancher/rancher/pkg/generated/controllers/provisioning.cattle.io/v1"
+	rkecontrollers "github.com/rancher/rancher/pkg/generated/controllers/rke.cattle.io/v1"
+	upgradev1 "github.com/rancher/rancher/pkg/generated/controllers/upgrade.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/namespace"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type handler struct {
+	mgmtClusterName      string
+	clusterCache         provisioningcontrollers.ClusterCache
+	downstreamAppClient  catalogv1.AppClient
+	downstreamPlanClient upgradev1.PlanClient
+}
+
+func Register(ctx context.Context, context *config.UserContext) {
+	h := handler{
+		mgmtClusterName:      context.ClusterName,
+		clusterCache:         context.Management.Wrangler.Provisioning.Cluster().Cache(),
+		downstreamAppClient:  context.Catalog.V1().App(),
+		downstreamPlanClient: context.Plan.V1().Plan(),
+	}
+
+	rkecontrollers.RegisterRKEControlPlaneStatusHandler(ctx, context.Management.Wrangler.RKE.RKEControlPlane(),
+		"", "sync-system-upgrade-controller-condition", h.syncSystemUpgradeControllerCondition)
+}
+
+// syncSystemUpgradeControllerCondition checks the status of the system-upgrade-controller app in the downstream cluster
+// and manages the SystemUpgradeControllerReady condition on the RKEControlPlane object
+func (h *handler) syncSystemUpgradeControllerCondition(obj *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus) (rkev1.RKEControlPlaneStatus, error) {
+	if obj == nil || obj.DeletionTimestamp != nil {
+		return status, nil
+	}
+	if obj.Spec.ManagementClusterName != h.mgmtClusterName {
+		return status, nil
+	}
+
+	targetVersion := settings.SystemUpgradeControllerChartVersion.Get()
+	if targetVersion == "" {
+		logrus.Warn("[rkecontrolplanecondition] the SystemUpgradeControllerChartVersion setting is not set")
+		capr.SystemUpgradeControllerReady.Reason(&status, fmt.Sprintf("the SystemUpgradeControllerChartVersion setting is not set"))
+		capr.SystemUpgradeControllerReady.Message(&status, "")
+		capr.SystemUpgradeControllerReady.False(&status)
+		return status, nil
+	}
+
+	if capr.SystemUpgradeControllerReady.IsTrue(&status) {
+		actual := capr.SystemUpgradeControllerReady.GetMessage(&status)
+		if actual == targetVersion {
+			// Skip if the target version of the app has been installed
+			return status, nil
+		} else if actual == "" {
+			// If SystemUpgradeControllerReady is true but its message is empty, this may occur in scenarios where Rancher
+			// is upgraded to 2.12.x, then rolled back to 2.11.x, and later re-upgraded to 2.12.x without restoring the local cluster.
+			// In such cases, the condition should be rest
+			capr.SystemUpgradeControllerReady.Reason(&status, fmt.Sprintf("reset the condition"))
```

Comment:
You should drop the annotation here too, but you wouldn't be able to use a generating status handler.

Reply:
In this case, since the Fleet bundles for the old apps exist ( they are created after Rancher is rolled back to v2.11.x), the managesystemagent handler will remove the annotation after uninstalling the fleet bundles. 
Please let me know if I miss any scenario.